src/main/java/org/jenkinsci/plugins/pipeline/milestone/MilestoneStepExecution.java

```diff
@@ -339,6 +335,29 @@ private static void save() {
         }
     }
 
+    /**
+     * Clean up tracked jobs on deleted.
+     */
+    @Extension
+    public static class CleanupJobsOnDelete extends ItemListener {
+
+        @Override
+        public void onDeleted(Item item) {
+            if (item instanceof Job) {
+                String jobName = item.getFullName();
+                Map<Integer, Milestone> job = getMilestonesByOrdinalByJob().get(jobName);
```

Comment:
After a recent upgrade I'm getting a `NullPointerException` here.

```java
java.lang.NullPointerException
	at org.jenkinsci.plugins.pipeline.milestone.MilestoneStepExecution$CleanupJobsOnDelete.onDeleted(MilestoneStepExecution.java:348)
	at hudson.model.listeners.ItemListener$4.apply(ItemListener.java:205)
	at hudson.model.listeners.ItemListener$4.apply(ItemListener.java:203)
	at hudson.model.listeners.ItemListener.forAll(ItemListener.java:167)
	at hudson.model.listeners.ItemListener.fireOnDeleted(ItemListener.java:203)
	at com.cloudbees.hudson.plugins.folder.AbstractFolder.onDeleted(AbstractFolder.java:949)
	at com.cloudbees.hudson.plugins.folder.AbstractFolder.onDeleted(AbstractFolder.java:135)
	at hudson.model.AbstractItem.delete(AbstractItem.java:593)
	at hudson.model.Job.delete(Job.java:683)
	at com.cloudbees.hudson.plugins.folder.computed.ComputedFolder.updateChildren(ComputedFolder.java:198)
	at com.cloudbees.hudson.plugins.folder.computed.FolderComputation.run(FolderComputation.java:139)
	at hudson.model.ResourceController.execute(ResourceController.java:98)
	at hudson.model.Executor.run(Executor.java:404)
```

My guess is that `getMilestonesByOrdinalByJob` is returning `null`, though I'm not sure why it would be.